File: Smallest SubArr With Sum/Brute-Force/bruteForce.ts
const smallestSubWithSum = (arr: number[], x: number) => {
  const n = arr.length;
  let res = Infinity;

  for (let i = 0; i < n; i++) {
    let sum = 0;
    for (let j = i; j < n; j++) {
      sum += arr[j];
      if (sum > x) {
        res = Math.min(res, j - i + 1);
        break;
      }
    }
  }
  if (res === Infinity) return 0;
  return res;
}

let arr = [1, 4, 45, 6, 10, 19];
let x = 51;

console.log(smallestSubWithSum(arr, x));

File: 2. Smallest SubArr With Sum/Brute-Force/bruteForce.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

int smallestSubWithSum(int x, vector<int>& arr) {
  int n = arr.size();
  int res = INT_MAX;

  for (int i = 0; i < n; i++) {
    int sum = 0;
    for (int j = i; j < n; j++) {
      sum += arr[j];

      if (sum > x) {
        res = min(res, j - i + 1);
        break;
      }
    }
  }

  if (res == INT_MAX) {
    return 0;
  }
  return res;
}

int main() {
    vector<int> arr = {1, 4, 45, 6, 10, 19};
    int x = 51;

    cout << smallestSubWithSum(x, arr);

    return 0;
}
